### PR TITLE
qemu.tests: Add virtio serial unplug test case and script.

### PR DESCRIPTION
**Pull request by FengYang**
Signed-off-by: Feng Yang <fyang@redhat.com>

### CODE REVIEW

**Comment by FengYang** [comment]
ID: 1157455  1157384

**Comment by CongSmile** [comment]
Test Failed:
04/15 12:37:02 ERROR|      virt:0171| Test failed: TestError: Fail to get device id from info qtree command. match string: dev: virtserialport, id \"(.*)\"\s*chardev = \"(.*)\"\s*nr = [0-9]* \(\w*\)\s*name = \"vc1\" Error message: list index out of rangeinfo qtree output:
...
        bus: virtio-bus^M
          type virtio-pci-bus^M
          dev: virtio-serial-device, id ""^M
            max_ports = 31^M
            bus: virtio_serial_pci0.0^M
              type virtio-serial-bus^M
              dev: virtserialport, id "vc2"^M
                chardev = devvc2^M
                nr = 2^M
                name = "vc2"^M
                port 2, guest off, host off, throttle off^M
              dev: virtserialport, id "vc1"^M
                chardev = devvc1^M
                nr = 1^M
                name = "vc1"^M
                port 1, guest off, host off, throttle off^M

File: qemu/tests/virtio_serial_hotunplug.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        match_string = sub_params.get("match_string", "dev: %s, id \"(.*)\"")
+        unplug_chardev = sub_params.get("unplug_chardev", "no") == "yes"
+        device_type = sub_params.get("device_type")
+        match_string = match_string % (device_type, device)
+        try:
+            device_id, chardev_id = re.findall(match_string, qtree_output)[0]
+        except Exception, err:
+            txt = "Fail to get device id from info qtree command. "
+            txt += "match string: %s Error message: %s" % (match_string, err)
+            txt += "info qtree output:\n%s" % qtree_output
```
Comment (CongSmile) [inline qemu/tests/virtio_serial_hotunplug.py]: Hi @FengYang 

Could we add '\n' between the error msg?
It's too long to read.

Thanks.

**Comment by CongSmile** [comment]
Hi @FengYang 

From the case, we need "on the host, open the chardev but don't read from it.",
seems there is no such step.

Not sure whether I missed it.

Could you help check it?

Thanks.

**Comment by FengYang** [comment]
Hi, Cong

2.on the host, open the chardev but don't read from it.
and step 3 just the file_transfer steps.

**Comment by XueqiangWei** [comment]
Test failed and same error as Cong

Error info:
TestError: Fail to get device id from info qtree command. match string: dev: virtserialport, id \"(.*)\"\s*chardev = \"(.*)\"\s*nr = [0-9]* \(\w*\)\s*name = \"vc1\" Error message: list index out of rangeinfo qtree output:
......


Xueqiang Wei

**Comment by xutian** [comment]
open new PR https://github.com/autotest/tp-qemu/pull/393 to do update, so close this one.

Thanks @FengYang

**Pull request closed without merging**